src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -118,27 +119,50 @@ public static void deleteTempTargetBQTable() throws IOException, InterruptedExce
         .append(UUID.randomUUID()).append("'), ");
     }
-    BigQueryClient.getSoleQueryResult("create table `test_automation." + gcsBqSourceTable + "` as " +
+    BigQueryClient.getSoleQueryResult("create table `test_automation." + bqSourceTable + "` as " +
                                         "SELECT * FROM UNNEST([ " +
                                         " STRUCT(1 AS Id, " + ((int) (Math.random() * (1000) + 1)) + " as Value, " +
                                         "'" + UUID.randomUUID() + "' as UID), " +
                                         records +
                                         "  (26, " + ((int) (Math.random() * (1000) + 1)) + ", " +
                                         "'" + UUID.randomUUID() + "') " +
                                         "])");
-    BeforeActions.scenario.write("BQ source Table " + gcsBqSourceTable + " created successfully");
+    BeforeActions.scenario.write("BQ source Table " + bqSourceTable + " created successfully");
   }
 
-  @After(order = 1, value = "@BQ_SOURCE_TEST")
+  @After(order = 1, value = "@BQ_SOURCE_TEST or @BQ_PARTITIONED_SOURCE_TEST")
   public static void deleteTempSourceBQTable() throws IOException, InterruptedException {
-    BigQueryClient.dropBqQuery(gcsBqSourceTable);
-    BeforeActions.scenario.write("BQ source Table " + gcsBqSourceTable + " deleted successfully");
+    BigQueryClient.dropBqQuery(bqSourceTable);
+    BeforeActions.scenario.write("BQ source Table " + bqSourceTable + " deleted successfully");
+  }
+
+  /**
+   * Create BigQuery partitioned table(transaction_id INT64, transaction_uid STRING, transaction_date DATE)
+   * containing random testdata.
+   * Sample row:
+   *   transaction_id | transaction_uid                       | transaction_date
+   *   1              | 51c76c5c-543c-4066-8032-f4870f9e9a0b  | 2022-01-31
+   */
+  @Before(order = 1, value = "@BQ_PARTITIONED_SOURCE_TEST")
+  public static void createTempPartitionedSourceBQTable() throws IOException, InterruptedException {
+    bqSourceTable = "E2E_TEST_SOURCE_" + new SimpleDateFormat("yyyy_MM_dd_HH_mm_ss").format(new Date());
+    try {
+      BigQueryClient.getSoleQueryResult("create table `test_automation." + bqSourceTable + "` " +
+                                          "(transaction_id INT64, transaction_uid STRING, transaction_date DATE ) " +
+                                          "PARTITION BY _PARTITIONDATE");
+      BigQueryClient.getSoleQueryResult("INSERT INTO `test_automation." + bqSourceTable + "` " +
+                                          "(transaction_id, transaction_uid, transaction_date) " +
+                                          "SELECT ROW_NUMBER() OVER(ORDER BY GENERATE_UUID()), GENERATE_UUID(), date " +
+                                          "FROM UNNEST(GENERATE_DATE_ARRAY('2022-01-01', current_date())) AS date");
+    } catch (NoSuchElementException e) {
+    }
```

Comment:
When would NoSuchElementException be thrown, and why do we want to ignore it? Can you add a comment?

Reply:
BQ Insert query does not return any record. Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException.